backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraNodeIpAndPort.java

```diff
@@ -0,0 +1,76 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra.init;
+
+import com.google.common.base.Preconditions;
+
+public class CassandraNodeIpAndPort {
+    public static final int DEFAULT_CASSANDRA_PORT = 9042;
+
+    public static CassandraNodeIpAndPort parseConfString(String ipAndPort) {
+        Preconditions.checkNotNull(ipAndPort);
+        Preconditions.checkArgument(!ipAndPort.isEmpty());
+
+        int port;
+        String[] parts = ipAndPort.split(":");
+
+        if(parts.length == 1) {
+            port = DEFAULT_CASSANDRA_PORT;
+        }
+        else if(parts.length == 2) {
+            port = Integer.parseInt(parts[1]);
+        }
+        else {
+            throw new IllegalArgumentException(ipAndPort + " is not a valid cassandra node");
+        }
+
+        String ip = parts[0];
+        return new CassandraNodeIpAndPort(ip, port);
+    }
+
+    private final String ip;
+    private final int port;
+
+    public CassandraNodeIpAndPort(String ip, int port) {
+        this.ip = ip;
+        this.port = port;
+    }
+
+    public CassandraNodeIpAndPort(String ip) {
+        this(ip, DEFAULT_CASSANDRA_PORT);
+    }
+
+    public String getIp() {
+        return ip;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    @Override
+    public String toString() {
+        if (DEFAULT_CASSANDRA_PORT != port) {
```

Comment:
I removed toString (see comment above for the reason I don't display port in all case)